### PR TITLE
pythonPackages.dash: init at 1.3.0

### PR DESCRIPTION
**Pull request by jeroendehaas**
This PR introduces Dash, a framework for building data-driven web applications. Dash relies on a number of related packages which this PR adds as well.

###### Things done

- [ ] Tested using sandboxing ([nix.useSandbox](http://nixos.org/nixos/manual/options.html#opt-nix.useSandbox) on NixOS, or option `sandbox` in [`nix.conf`](http://nixos.org/nix/manual/#sec-conf-file) on non-NixOS)
- Built on platform(s)
   - [x] NixOS
   - [ ] macOS
   - [ ] other Linux distributions
- [ ] Tested via one or more NixOS test(s) if existing and applicable for the change (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
- [ ] Tested compilation of all pkgs that depend on this change using `nix-shell -p nix-review --run "nix-review wip"`
- [ ] Tested execution of all binary files (usually in `./result/bin/`)
- [ ] Determined the impact on package closure size (by running `nix path-info -S` before and after)
- [ ] Ensured that relevant documentation is up to date
- [x] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/.github/CONTRIBUTING.md).

### CODE REVIEW

File: pkgs/development/python-modules/dash-core-components/default.nix [hunk 1]
```diff
@@ -0,0 +1,27 @@
...
+, fetchPypi
+}:
+
+buildPythonPackage rec {
+  pname = "dash-core-components";
+  version = "1.2.0";
+
+  src = fetchPypi {
+    inherit version;
+    pname = "dash_core_components";
```
Comment (alexarice) [inline pkgs/development/python-modules/dash-core-components/default.nix]: Could just have `inherit version pname;`

File: pkgs/development/python-modules/dash-core-components/default.nix [hunk 2]
```diff
@@ -0,0 +1,27 @@
...
+    sha256 = "910c158081c337c56c6d0a3f617a5bdb275933ffa513ea4ed2448c682650edf6";
+  };
+
+  doCheck = false; # module tests depend on dash, dash depends on this package
+
+  propagatedBuildInputs = [];
```
Comment (alexarice) [inline pkgs/development/python-modules/dash-core-components/default.nix]: Does this do anything?

File: pkgs/development/python-modules/dash-core-components/default.nix [hunk 3]
```diff
@@ -0,0 +1,27 @@
...
+
+  meta = with stdenv.lib; {
+    description = "A Dash component starter pack.";
```
Comment (alexarice) [inline pkgs/development/python-modules/dash-core-components/default.nix]: meta description should not end with a .

File: pkgs/development/python-modules/dash-core-components/default.nix [hunk 4]
```diff
@@ -0,0 +1,27 @@
...
+    inherit version;
+    pname = "dash_core_components";
+    sha256 = "910c158081c337c56c6d0a3f617a5bdb275933ffa513ea4ed2448c682650edf6";
+  };
+
+  doCheck = false; # module tests depend on dash, dash depends on this package
+
+  propagatedBuildInputs = [];
+
+  meta = with stdenv.lib; {
```
Comment (alexarice) [inline pkgs/development/python-modules/dash-core-components/default.nix]: Could have `with lib;` and change `stdenv` to `lib` at the top

File: pkgs/development/python-modules/dash-html-components/default.nix [hunk 5]
```diff
@@ -0,0 +1,27 @@
...
+    inherit version;
+    pname = "dash_html_components";
+    sha256 = "7d7e80bf12635eb7d281fb0e7dd4b6e53f10dd57e3384ea5044c44ab16454fe4";
+  };
+
+  doCheck = false; # depends on dash which depends on this package
+
+  propagatedBuildInputs = [];
+
+  meta = with stdenv.lib; {
```
Comment (alexarice) [inline pkgs/development/python-modules/dash-html-components/default.nix]: same here

File: pkgs/development/python-modules/dash-renderer/default.nix [hunk 6]
```diff
@@ -0,0 +1,27 @@
...
+    inherit version;
+    pname = "dash_renderer";
+    sha256 = "eca9a3ee9604f349912aa7956cdce547378937d9e9491d3bf02f719cfea278e0";
+  };
+
+  doCheck = false; # depends on dash which depends on this package
+
+  propagatedBuildInputs = [];
+
+  meta = with stdenv.lib; {
```
Comment (alexarice) [inline pkgs/development/python-modules/dash-renderer/default.nix]: and here

File: pkgs/development/python-modules/dash-table/default.nix [hunk 7]
```diff
@@ -0,0 +1,27 @@
...
+    inherit version;
+    pname = "dash_table";
+    sha256 = "ca6855dcf95136f19e0ab273c48caf8a5529f2a614bbc5ae02c04e5d0284db33";
+  };
+
+  doCheck = false; # module tests depend on dash, dash depends on this package
+
+  propagatedBuildInputs = [];
+
+  meta = with stdenv.lib; {
```
Comment (alexarice) [inline pkgs/development/python-modules/dash-table/default.nix]: and here

File: pkgs/development/python-modules/dash/default.nix [hunk 8]
```diff
@@ -0,0 +1,44 @@
...
+    dash-table
+    future
+    flask
+    flask-compress
+    plotly
+  ];
+
+  doCheck = false; # no tests included
+
+  meta = with stdenv.lib; {
```
Comment (alexarice) [inline pkgs/development/python-modules/dash/default.nix]: and here

File: pkgs/development/python-modules/dash-table/default.nix [hunk 9]
```diff
@@ -0,0 +1,27 @@
...
+
+  src = fetchPypi {
+    inherit version;
+    pname = "dash_table";
+    sha256 = "ca6855dcf95136f19e0ab273c48caf8a5529f2a614bbc5ae02c04e5d0284db33";
+  };
+
+  doCheck = false; # module tests depend on dash, dash depends on this package
+
+  propagatedBuildInputs = [];
```
Comment (alexarice) [inline pkgs/development/python-modules/dash-table/default.nix]: same comment about empty propagatedBuildInputs

File: pkgs/development/python-modules/dash-table/default.nix [hunk 10]
```diff
@@ -0,0 +1,27 @@
...
+, fetchPypi
+}:
+
+buildPythonPackage rec {
+  pname = "dash-table";
+  version = "4.3.0";
+
+  src = fetchPypi {
+    inherit version;
+    pname = "dash_table";
```
Comment (alexarice) [inline pkgs/development/python-modules/dash-table/default.nix]: and with `inherit pname;`

File: pkgs/development/python-modules/dash/default.nix [hunk 11]
```diff
@@ -0,0 +1,44 @@
...
+    dash-html-components
+    dash-renderer
+    dash-table
+    future
+    flask
+    flask-compress
+    plotly
+  ];
+
+  doCheck = false; # no tests included
```
Comment (alexarice) [inline pkgs/development/python-modules/dash/default.nix]: is it possible to somehow include the tests for the other packages?

**Review by alexarice — COMMENTED** [review]
Most of these are just repeats across all the files

File: pkgs/development/python-modules/dash-core-components/default.nix (hunk 1 above)
Comment (alexarice) [reply pkgs/development/python-modules/dash-core-components/default.nix]: ah, I see, my bad

File: pkgs/development/python-modules/dash-core-components/default.nix (hunk 2 above)
Comment (jeroendehaas) [reply pkgs/development/python-modules/dash-core-components/default.nix]: This module does not specify any packages in its `install_requires` list. It seems rather self-contained (if you ignore the circular dependency between dash and dash-core-components)

Comment (alexarice) [reply pkgs/development/python-modules/dash-core-components/default.nix]: My point was more that you could remove the whole line

Comment (jeroendehaas) [reply pkgs/development/python-modules/dash-core-components/default.nix]: Agreed. These have been removed

File: pkgs/development/python-modules/dash/default.nix (hunk 11 above)
Comment (jeroendehaas) [reply pkgs/development/python-modules/dash/default.nix]: Upon further inspection it seems the packages do not include any install tests (which causes the test phase to fail), so I've updated the comment

**Comment by t184256** [comment]
Seems like it's obsoleted by https://github.com/NixOS/nixpkgs/pull/77837

**Comment by jonringer** [comment]
dash-1.9.1 is available on master

```
[09:47:27] jon@jon-desktop /home/jon/projects/nixpkgs (master)
$ nix-build -A python3Packages.dash
/nix/store/cjs2n1hj31f9ag824w2nkj1a4m91w7dv-python3.7-dash-1.9.1
```

**Pull request closed without merging**